Libplanet/Net/Swarm.cs

```diff
@@ -718,8 +724,14 @@ void ISwarm.ReplyMessage(Message message)
 
                 if (!blockDownloadComplete)
                 {
-                    _logger.Information("Failed to download blocks from peers.");
-                    return;
+                    if (PreloadBlockDownloadFailed is null)
+                    {
+                        throw new SwarmException("Failed to download blocks from peers.");
```

Comment:
It would be better if we have a more specific subtype for this exception.

Reply:
Changed to throwing 'AggregateException' which has exceptions thrown during block download.